dotnet/src/Connectors/Connectors.AI.OpenAI/ChatCompletionWithData/AzureOpenAIChatCompletionWithData.cs

```diff
@@ -59,81 +59,29 @@ public ChatHistory CreateNewChat(string? instructions = null)
+        => this.InternalGetChatStreamingContentsAsync(chatHistory, executionSettings, kernel, cancellationToken);
 
     /// <inheritdoc/>
-    public async Task<IReadOnlyList<ITextResult>> GetCompletionsAsync(
-        string prompt,
-        PromptExecutionSettings? executionSettings = null,
-        Kernel? kernel = null,
-        CancellationToken cancellationToken = default)
+    public async Task<IReadOnlyList<TextContent>> GetTextContentsAsync(string prompt, PromptExecutionSettings? executionSettings = null, Kernel? kernel = null, CancellationToken cancellationToken = default)
     {
-        OpenAIPromptExecutionSettings chatExecutionSettings = OpenAIPromptExecutionSettings.FromRequestSettingsWithData(executionSettings);
-
-        var chat = InternalCreateNewChat(prompt, chatExecutionSettings);
-
-        return (await this.GetChatCompletionsAsync(chat, chatExecutionSettings, kernel, cancellationToken).ConfigureAwait(false))
-            .OfType<ITextResult>()
-            .ToList();
+        return (await this.GetChatContentsAsync(prompt, executionSettings, kernel, cancellationToken).ConfigureAwait(false)).Select(chat => new TextContent(chat.Content, chat, Encoding.UTF8, chat.Metadata)).ToList();
     }
 
     /// <inheritdoc/>
-    public IAsyncEnumerable<T> GetStreamingContentAsync<T>(
+    public async IAsyncEnumerable<StreamingTextContent> GetStreamingTextContentsAsync(
         string prompt,
         PromptExecutionSettings? executionSettings = null,
         Kernel? kernel = null,
-        CancellationToken cancellationToken = default)
-    {
-        OpenAIPromptExecutionSettings chatExecutionSettings = OpenAIPromptExecutionSettings.FromRequestSettingsWithData(executionSettings);
-
-        var chat = InternalCreateNewChat(prompt, chatExecutionSettings);
-
-        return this.GetStreamingContentAsync<T>(chat, chatExecutionSettings, kernel, cancellationToken);
-    }
-
-    /// <inheritdoc/>
-    public async IAsyncEnumerable<T> GetStreamingContentAsync<T>(
-        ChatHistory chatHistory,
-        PromptExecutionSettings? executionSettings = null,
-        Kernel? kernel = null,
         [EnumeratorCancellation] CancellationToken cancellationToken = default)
     {
-        OpenAIPromptExecutionSettings chatRequestSettings = OpenAIPromptExecutionSettings.FromRequestSettingsWithData(executionSettings);
-
-        using var request = this.GetRequest(chatHistory, chatRequestSettings, isStreamEnabled: true);
-        using var response = await this.SendRequestAsync(request, cancellationToken).ConfigureAwait(false);
-
-        await foreach (var result in this.GetChatStreamingUpdatesAsync<T>(response))
+        await foreach (var streamingChatContent in this.InternalGetChatStreamingContentsAsync(new ChatHistory(prompt), executionSettings, kernel, cancellationToken).ConfigureAwait(false))
```

Comment:
Constructing a new ChatHistory from the prompt here is ok, rather than going through the parsing logic used elsewhere?

Reply:
Could we make the services (i.e., `IChatCompletion`, `ITextCompletion`, and `IQuestionAndAnswering`) generics?
- `IChatCompletion` -> `IChatMessageOutput<TInput>`
- `ITextCompletion` -> `ITextOutput<TInput>`
- `IQuestionAndAnswering` -> `IQuestionAndAnsweringOutput<TInput>`

The interfaces would then look something like...

```csharp
public interface ITextCompletion<T>
{
    Task<IReadOnlyList<TextContent>> GetTextContentsAsync(
        T input,
        PromptExecutionSettings? executionSettings = null,
        Kernel? kernel = null,
        CancellationToken cancellationToken = default);
}
```

The connectors could then implement them with whatever inputs they expect. For example, `OpenAIChatCompletion` would require a `ChatHistory` object.

```csharp
public interface OpenAIChatCompletion: IChatMessageOutput<ChatHistory>, ITextOutput<ChatHistory>, IQuestionAndAnsweringOutput<ChatHistory>
{
   Task<IReadOnlyList<TextContent>> GetTextContentsAsync(
       ChatHistory history,
       PromptExecutionSettings? executionSettings = null,
       Kernel? kernel = null,
       CancellationToken cancellationToken = default)
   {
       // Implementation
   }
}
```

The extensions would then need to provide default behavior for each of the input types

```csharp
public static class TextCompletionExtensions
{
    // Extension for string input type
    public static async Task<TextContent> GetTextContentAsync(
        this ITextCompletion<string> chatCompletion,
        string prompt,
        PromptExecutionSettings? executionSettings = null,
        Kernel? kernel = null,
        CancellationToken cancellationToken = default)
        => (await chatCompletion.GetTextContentsAsync(prompt, executionSettings, kernel, cancellationToken).ConfigureAwait(false))
            .Single();
    
    // Extension for ChatHistory input type
    public static async Task<TextContent> GetTextContentAsync(
        this ITextCompletion<ChatHistory> chatCompletion,
        string prompt,
        PromptExecutionSettings? executionSettings = null,
        Kernel? kernel = null,
        CancellationToken cancellationToken = default)
    {
        // massage prompt into a ChatHistory object
        return await chatCompletion.GetTextContentsAsync(chatHistory, executionSettings, kernel, cancellationToken).ConfigureAwait(false))
            .Single();
}
```